# TEST 1 : ANOVA PAR CENTRE (H1)
Objectif : montrer que les centres n’ont pas la même charge moyenne.

In [3]:
import pandas as pd
import scipy.stats as stats

df = pd.read_excel(r"C:\Users\Tenordem\Documents\projet-repartition-charge\data\raw\data_complet_v2.xlsx")

In [6]:
df[["Centre", "ChargeTotaleActualisee"]].head()


,Centre,ChargeTotaleActualisee
0,-,2300
1,-,2300
2,-,2300
3,-,2300
4,CDAD-BDX,2200


In [12]:
groupes = [
    g["ChargeTotaleActualisee"]
    for _, g in df_clean.groupby("Centre")
    if g["ChargeTotaleActualisee"].notna().sum() > 1  # éviter les groupes vides
]


In [13]:
from scipy.stats import f_oneway

F_stat, p_value = f_oneway(*groupes)
F_stat, p_value


(np.float64(29.650390600369576), np.float64(2.4335087603650404e-35))

## conclusion
Le test ANOVA appliqué à la charge totale actualisée par centre montre une différence statistiquement significative entre les centres (F = 29.65, p < 0.001).
Cela indique que les centres ne présentent pas le même niveau de charge moyenne.
L’hypothèse H1 (“certains centres sont plus chargés que d’autres”) est donc validée.

# TEST 2 — ANOVA : ChargeTotaleActualisee ~ Types_phases

Objectif : Vérifier si les phases ont des charges moyennes différentes.

In [14]:
df[['Types_phases', 'ChargeTotaleActualisee']].head()


,Types_phases,ChargeTotaleActualisee
0,AUTRES,2300.0
1,AUTRES,2300.0
2,AUTRES,2300.0
3,AUTRES,2300.0
4,REALISATION,2200.0


In [16]:
groupes_phases = [
    g["ChargeTotaleActualisee"]
    for _, g in df.groupby("Types_phases")
    if g["ChargeTotaleActualisee"].notna().sum() > 1
]


In [17]:
from scipy.stats import f_oneway

F_stat2, p_value2 = f_oneway(*groupes_phases)
F_stat2, p_value2


(np.float64(1301.19128225415), np.float64(0.0))

# conclusion 
Le test ANOVA réalisé sur la charge totale actualisée selon les différentes phases montre une différence statistiquement significative entre les moyennes des phases (F = 1301.19, p < 0.001).
Cela confirme que les phases ne consomment pas du tout la même quantité de charge.

Plus précisément, la phase Réalisation représente de loin la charge la plus élevée, suivie par les phases de Maintenance et d’Étude.

L’hypothèse H2 — “Certaines phases consomment beaucoup plus de ressources que d’autres” — est donc validée.

# TEST 3 — Corrélation Pearson : ChargeTotaleEstimee vs ChargeTotaleActualisee

In [18]:
df["ChargeTotaleEstimee"] = pd.to_numeric(df["ChargeTotaleEstimee"], errors="coerce")
df["ChargeTotaleActualisee"] = pd.to_numeric(df["ChargeTotaleActualisee"], errors="coerce")


In [19]:
df_corr = df.dropna(subset=["ChargeTotaleEstimee", "ChargeTotaleActualisee"])


In [20]:
from scipy.stats import pearsonr

r, p_value = pearsonr(df_corr["ChargeTotaleEstimee"], df_corr["ChargeTotaleActualisee"])
r, p_value


(np.float64(0.8901056270940687), np.float64(0.0))

# conclusion
Il existe une corrélation très forte (r = 0.89) et statistiquement significative (p < 0.001) entre la charge estimée et la charge réellement exécutée.
Les estimations de charge sont globalement fiables et suivent très bien la charge réelle, même si des écarts peuvent exister au niveau des centres ou des modules.